sdk/go/pulumi/policyx/schema.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policyx
+
+type EnforcementLevel int32
+
+const (
+	EnforcementLevelAdvisory  EnforcementLevel = 0 // Displayed to users, but does not block deployment.
+	EnforcementLevelMandatory EnforcementLevel = 1 // Stops deployment, cannot be overridden.
+	EnforcementLevelDisabled  EnforcementLevel = 2 // Disabled policies do not run during a deployment.
+)
+
+type PolicyConfigJSONSchemaTypes []PolicyConfigJSONSchemaType
+
+type PolicyConfigJSONSchemaType string
+
+const (
+	PolicyConfigJSONSchemaTypeBoolean PolicyConfigJSONSchemaType = "boolean"
+	PolicyConfigJSONSchemaTypeNumber  PolicyConfigJSONSchemaType = "number"
+	PolicyConfigJSONSchemaTypeNull    PolicyConfigJSONSchemaType = "null"
+	PolicyConfigJSONSchemaTypeObject  PolicyConfigJSONSchemaType = "object"
+	PolicyConfigJSONSchemaTypeString  PolicyConfigJSONSchemaType = "string"
+)
+
+type PolicyConfigJSONSchemaTypeName string
+
+const (
+	PolicyConfigJSONSchemaTypeNameString  PolicyConfigJSONSchemaTypeName = "string"
+	PolicyConfigJSONSchemaTypeNameNumber  PolicyConfigJSONSchemaTypeName = "number"
+	PolicyConfigJSONSchemaTypeNameInteger PolicyConfigJSONSchemaTypeName = "integer"
+	PolicyConfigJSONSchemaTypeNameBoolean PolicyConfigJSONSchemaTypeName = "boolean"
+	PolicyConfigJSONSchemaTypeNameObject  PolicyConfigJSONSchemaTypeName = "object"
+	PolicyConfigJSONSchemaTypeNameArray   PolicyConfigJSONSchemaTypeName = "array"
+	PolicyConfigJSONSchemaTypeNameNull    PolicyConfigJSONSchemaTypeName = "null"
+)
+
+type PolicyConfigJSONSchema struct {
```

Comment:
Be good to at least doc comment the type I reckon even if we can't do all the fields right now